lockfile_generator/src/cargo.rs

```diff
@@ -1,15 +1,24 @@
 //! Rust cargo ecosystem.
 
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::process::Command;
 
-use crate::Generator;
+use rust_cargo::core;
+use rust_cargo::util::config::Config;
+
+use crate::{Error, Generator, Result};
 
 pub struct Cargo;
 
 impl Generator for Cargo {
-    fn lockfile_name(&self) -> &'static str {
-        "Cargo.lock"
+    fn lockfile_path(&self, manifest_path: &Path) -> Result<PathBuf> {
+        let cargo_config = Config::default()?;
+        let root_manifest =
+            core::find_workspace_root(manifest_path, &cargo_config)?.unwrap_or_default();
```

Comment:
Does `find_workspace_root` also work when there is no workspace specified in the manifest?

Reply:
I thought it would, considering that's how `cargo metadata` works. But this will indeed be `None` when there is no workspace configured. I fixed it to fall back to the manifest path itself.